server/src/main/java/com/broadcom/lsp/cobol/core/messages/JsonMessageServiceImpl.java

```diff
@@ -0,0 +1,154 @@
+      "Keys with id '{%s}' already exists.";
+  private static final String RESOURCE_LOAD_EXCEPTION_MESSAGE =
+      "Please load messages from a template json file before using message service."
+          + "Use loadMessages method provided by the MessageService class.";
+  private static final String KEY_NOT_FOUND_MESSAGE =
+      "No templates corresponding to key '{}' found.";
+  private final Map<String, MessageTemplate> customMessagesMap = new HashMap<>();
+
+  public JsonMessageServiceImpl(String fileName, Optional<Locale> locale) {
+    loadMessages(fileName, locale);
+  }
+
+  /***
+   *  This method return an {@link ExternalizedMessage} object passed on passed params from an message file.
+   * @param key Key corresponds to the uniqueness of a messageTemplate.
+   *            It's like an ID for the message. Every key must be unique.
+   * @param parameters are the different arguments to be substituted in the message,
+   *            in the passed order.
+   * @return {@link ExternalizedMessage}
+   */
+  @Override
+  public ExternalizedMessage getMessage(@NonNull String key, @NonNull Object... parameters) {
+    if (this.customMessagesMap.isEmpty()) {
+      throw new MessageTemplateLoadException(RESOURCE_LOAD_EXCEPTION_MESSAGE);
+    }
+    MessageTemplate messageTemplate = validateMessageTemplateExists(key);
+    if (!messageTemplate.isValidKey()) {
+      parameters = new Object[] {key};
+    }
+    return ExternalizedMessage.builder()
+        .messageTemplate(messageTemplate)
+        .parameters(parameters)
+        .key(key)
+        .build();
+  }
+
+  private MessageTemplate validateMessageTemplateExists(String key) {
+    return this.getMessageTemplate(key)
+        .orElseGet(
+            () -> {
+              LOG.error(KEY_NOT_FOUND_MESSAGE, key);
+              return invalidMessageTemplate();
+            });
+  }
+
+  private MessageTemplate invalidMessageTemplate() {
+    return new MessageTemplate(INVALID_KEY, INVALID_KEY_EXTERNALIZED_MESSAGE);
+  }
+
+  /***
+   * This methods verify and loads all messages from a provided message file.
+   * @param filename is the path from where we want to load messageTemplate files.
+   * @Throws {@link MissingResourceException}
+   */
+  @Override
+  public void loadMessages(String filename, Optional<Locale> locale) {
+    ResourceBundle.Control control = new JSONResourceBundleControl();
+    ResourceBundle customResourceBundle =
+        ResourceBundle.getBundle(
+            filename, locale.orElse(new Locale(DEFAULT_LOCALE, DEFAULT_LOCALE)), control);
+    List<MessageTemplate> messageTemplateList =
+        Collections.list(customResourceBundle.getKeys()).stream()
+            .map(e -> (MessageTemplate) customResourceBundle.getObject(e))
+            .collect(Collectors.toList());
+    this.addMessageTemplates(messageTemplateList);
+  }
+
+  /***
+   * Adds a list of {@link MessageTemplate} to an existing object of {@link MessageService }
```

Comment:
The pattern for the method Javadoc is that it should start with a verb in the present tense, active voice

Reply:
Planning to write - "Add a list of  ... ". I seriously don't have a hold on English grammar :D